packages/components/src/components/post-collapsible/post-collapsible.tsx

```diff
@@ -0,0 +1,115 @@
+import { Component, Element, h, Method, Prop, State } from '@stencil/core';
+import { checkValidHTMLTag, getElementHeight, onTransitionEnd } from '../../utils';
+
+let nextId = 0;
+
+@Component({
+  tag: 'post-collapsible',
+  styleUrl: 'post-collapsible.scss',
+  shadow: true,
+})
+export class PostCollapsible {
+  /**
+   * If `true`, the element is initially collapsed otherwise it is displayed.
+   */
+  @Prop({ mutable: true }) collapsed?: boolean = false;
+
+  /**
+   * Defines the hierarchical level of the collapsible header within the headings structure.
+   */
+  @Prop() headingLevel?: number = 2;
+
+  @State() collapseClasses: string;
+  @State() collapseHeight: string | null = null;
+
+  @Element() host: HTMLElement;
+
+  hasHeader: boolean;
+  headingTag: string | undefined;
+  collapsibleId: string;
+  collapsibleElement: HTMLElement;
+
+  componentWillLoad() {
+    this.hasHeader = !!this.host.querySelector('[slot="header"]');
+    if (!this.hasHeader) {
+      console.warn('Be sure to bind the post-collapsible to its control using aria-controls and aria-expanded attributes. More information here: https://getbootstrap.com/docs/5.2/components/collapse/#accessibility');
+    }
+
+    this.headingTag = checkValidHTMLTag(`h${this.headingLevel}`, 'The post-collapsible element requires a heading level between 1 and 6.');
```

Comment:
This check will produce a wrong positiv value, if someone enters "-1", "-2", etc. as heading-level property.
Because "h-1" is basically a correct html-tag.

Reply:
It is then probably overkill to just check if a string is contained in a list. What do you think of [this](https://github.com/swisspost/design-system/pull/683/commits/73d6d1fa387b5f376cb108c5b04641f8c6728e89) instead?

Reply:
I just realized, that I criticized the wrong line of code.
What we really should do is to validate the Property before using it!
I found how to validate props in stencil here: https://stenciljs.com/docs/properties#prop-validation, so...

```javascript
export class PostCollapsible {
  @Prop({ mutable: true }) headingLevel?: number = 2;
  
  ...

  this.headingTag = `h${this.headingLevel}`;

  ...
  
  // if we could abstract this in a utility-function, that would be great
  @Watch('headingLevel')
  headingLevelValidation(newValue: string, _oldValue: string) {
    if (![1, 2, 3, 4, 5, 6].includes(newValue)) {
      throw new Error('The post-collapsible element requires a heading level between 1 and 6.');
    }
  }
};
```

For the given case it should not be necessary to use a utility like "checkValidHTMLTag" anymore after this.

Reply:
I had seen this page, however in the storybook the component is reloaded each time you modify an argument and since the @Watch decorator is not called on load, the error is never displayed.

Please check [my last commit](https://github.com/swisspost/design-system/pull/683/commits/a6949b0421069f8c27b0fe045ad3d8e17161d3a0) where I mixed heading level check with @Watch and with componentWillLoad hook.